### ISSUE TITLE
[Feature proposal] Allow batch size in validation larger tran 1

### ISSUE THREAD

**Issue by zhiqwang** [xingyizhou/CenterNet#518]
### Question
In training, the batch size in validation dataset are restricted to be 1 in the master branch, see the code snippets below:
https://github.com/xingyizhou/CenterNet/blob/8ef87b433529ac8f8bd4f95707f6bc05052c55e9/src/main.py#L44-L50
It seems it is the size of `gt_det` are not same across the images in one batch, so it can not be collated using pytorch `collate_fn` function.

### Proposal
I would like to propose the following change in the size of `gt_det` that would allow pytorch's `collate_fn` function to support the batch size in validation dataset larger than 1.

### Detail
Because `reg_mask` records where the label annotation are, we can initialize `gt_det` as fixed size, and then obtain `gt_det` from index of `reg_mask`. Following are the code:
```python
gt_det = np.zeros((self.max_objs, 6), dtype=np.float32)
for k in range(num_objs):
    gt_det[k] = [ct[0] - w / 2, ct[1] - h / 2,
                 ct[0] + w / 2, ct[1] + h / 2, 1, cls_id]
```
From my restricted knowledge of this repo, `gt_det` are used in the debug function in trains, 
https://github.com/xingyizhou/CenterNet/blob/819e0d0dde02f7b8cb0644987a8d3a370aa8206a/src/lib/trains/ctdet.py#L93
and `gt_det` can be obtained using
```python
dets_gt = batch['meta']['gt_det'][batch['mask'] == 1].numpy().reshape(1, -1, dets.shape[2])
```

I think this would accelerate the training of doing this. Let me know if you think this is worth pursuing.

**Comment by xingyizhou** [comment]
Thank you for your proposal. However I usually disable validation in my training since the loss does not indicate anything.
